freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/model/common/Status.java

```diff
@@ -33,10 +41,10 @@
     public static final Collection<Status> AVAILABLE_STATUSES = List.of(AVAILABLE, MAINTENANCE_MODE_ENABLED);
```

Comment:
shouldn't the stack be available if it has the `UPSCALE_COMPLETED`, `DOWNSCALE_COMPLETED`, or `REPAIR_COMPLETED` status? There are places we check whether a stack is available before performing an operation on it.

Reply:
I would stick with `AVAILABLE` and won't introduce new states representing the same. What's the benefit of this?

Reply:
Hi @handavid and @lacikaaa , thanks for the review!

The reason we are changing the finishing status of a repair operation is because an operation will always change the status to running/available even operation failed or freeipa is unhealthy. So we will have a period where the available status is falsely reported.

<img width="607" alt="Screen Shot 2020-10-12 at 10 41 31 AM" src="https://user-images.githubusercontent.com/67844271/95766082-5bc24800-0c78-11eb-8661-8db2d25e485e.png">

One example,
Issue a reboot/repair change
<img width="593" alt="Screen Shot 2020-10-12 at 10 50 01 AM" src="https://user-images.githubusercontent.com/67844271/95766364-c8d5dd80-0c78-11eb-866e-3066ab889633.png">

Then instances are terminated so reboot/repair operation failed. But the operation will update status to running.  
<img width="578" alt="Screen Shot 2020-10-12 at 10 49 52 AM" src="https://user-images.githubusercontent.com/67844271/95766475-f0c54100-0c78-11eb-9c3d-ee973bc64b9c.png">

After 3 minutes, the actually health check find out freeipa is not running. 
<img width="604" alt="Screen Shot 2020-10-12 at 10 53 18 AM" src="https://user-images.githubusercontent.com/67844271/95766659-36820980-0c79-11eb-8b59-7e65bc43cf4d.png">

With this fix, the false running status will not be reported before a health check.

Reply:
ok so we would let the statuschecker to determine the status? in case of failed this seems right to me. Although it's a bit strange if we run repair/upscale/downscale and the flow finishes successfully and we can't expect the cluster is in a healthy state. It means the flow exit criteria isn't checking if it's job is done well. This just seems like a workaround to me, instead of fixing the flow (if it needs fixing)
cc @jamisonbennett